### PR TITLE
Created AnimesOnline presence

### LINKED ISSUE #4082: Animes Online | animesonline.org
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
Fushi#6666
**What's the name of the service?** 
animesonline.org | Animes Online
**What should the Presence display?** 
Show and episode you are watching, time elapsed/left

**If possible, please provide a logo for the service (512x512 minimum)** 
favicon

### PR DESCRIPTION
**Pull request by Faelayis**
Created AnimesOnline presence
- Added setting buttons
- Added setting Show timestamps

<details>
<summary>
Preview
</summary>
<br>

imageimageimageimageimageimage

</details>

resolves #4082

### CODE REVIEW

File: websites/A/AnimesOnline/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "color": "#e74c3c",
+  "tags": [
+    "animesonline",
+    "videos",
+    "anime"
+  ],
+  "url": "animesonline.org",
+  "description": {
+    "en": "Animes Online in FULLHD, HD and SD, choose the anime of your choice and watch it subtitled or dubbed! Join now it's free!",
+    "pt_BR": "Site de animes legendado em português. Assista em FULLHD, HD e SD.!"
```
Comment (Bas950) [inline websites/A/AnimesOnline/dist/metadata.json:20]:
```suggestion
    "pt_BR": "Animes Online em FULLHD, HD e SD, escolha o anime de sua preferencia e assista legendado ou dublado! Acesse agora é gratuito!"
```

File: websites/A/AnimesOnline/iframe.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+  let video: HTMLVideoElement;
+
+  if (document.querySelector(".jw-video") !== null)
+    video = document.querySelector(".jw-video");
+  else if (document.querySelector(".html5-video-container") !== null)
+    video = document.querySelector(".html5-video-container > video");
```
Comment (Bas950) [inline websites/A/AnimesOnline/iframe.ts:9]:
```suggestion
  const video: HTMLVideoElement = document.querySelector(".jw-video") || document.querySelector(".html5-video-container");
```
This should work but test if it actually does!

File: websites/A/AnimesOnline/presence.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      const titlemovie = playfilmes;
+      filmes = titlemovie;
+      filmes = `${filmes}`;
+      presenceData.details = "Filmes";
+      presenceData.state = filmes;
+    } else {
+      let info;
+      episode = "Assistir Anime";
+      presenceData.details = info;
+      presenceData.state = episode;
```
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:101]: info is not set?

File: websites/A/AnimesOnline/presence.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    );
+    if (playvdo.includes("Episódio")) {
+      const info = playvdo.split("- Episódio" || "Episódio");
+      episode = info.pop();
+      episode = `Episódio ${episode}`;
+      [presenceData.details] = info;
+      presenceData.state = episode;
+    } else if (titlemovie.includes("Sinopse")) {
+      const titlemovie = playfilmes;
+      filmes = titlemovie;
```
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:93]: seems like a useless extra const?

File: websites/A/AnimesOnline/presence.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    const timestamps = presence.getTimestamps(
+      Math.floor(video.current),
+      Math.floor(video.duration)
+    );
+    if (playvdo.includes("Episódio")) {
+      const info = playvdo.split("- Episódio" || "Episódio");
+      episode = info.pop();
+      episode = `Episódio ${episode}`;
+      [presenceData.details] = info;
+      presenceData.state = episode;
```
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:90]:
```suggestion
      [presenceData.details] = info;
      presenceData.state = `Episódio ${info.pop()}`;
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
Please make your code a bit cleaner, it looks like it uses a lot of useless variables.

**Comment by Faelayis** [comment]
> Please make your code a bit cleaner, it looks like it uses a lot of useless variables.
Sorry, I tested so much that I forgot to pay attention.

File: websites/A/AnimesOnline/presence.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  Search =
+    document.querySelector(
+      "#contenedor > div.module > div.content.rigth.csearch > header > h1"
+    )?.textContent ?? "pesquisa não encontrada",
+  playfilmes =
+    document.querySelector(
+      "#single > div.content.right > div.sheader > div.data > h1"
+    )?.textContent ?? "desconhecido",
+  playvdo = document.querySelector("#info > h1")?.textContent ?? "desconhecido",
+  path = document.location;
```
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:29]: there should be inside updatedata

File: websites/A/AnimesOnline/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  "color": "#e74c3c",
+  "tags": [
+    "animesonline",
+    "videos",
+    "anime"
+  ],
+  "url": "animesonline.org",
+  "description": {
+    "en": "Animes Online in FULLHD, HD and SD, choose the anime of your choice and watch it subtitled or dubbed! Join now it's free!",
+    "pt_BR": "Animes Online em FULLHD, HD e SD, escolha o anime de sua preferencia e assista legendado ou dublado! Acesse agora é gratuito!!"
```
Comment (Bas950) [inline websites/A/AnimesOnline/dist/metadata.json:20]:
```suggestion
    "pt_BR": "Animes Online em FULLHD, HD e SD, escolha o anime de sua preferencia e assista legendado ou dublado! Acesse agora é gratuito!"
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AnimesOnline/presence.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  "iFrameData",
+  (data: { current: number; duration: number; paused: boolean }) => {
+    video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
+  const time = await presence.getSetting("timestamps"),
+    buttons = await presence.getSetting("buttons"),
+    browsingStamp = Math.floor(Date.now() / 1000),
```
Comment (Slowlife01) [inline websites/A/AnimesOnline/presence.ts:26]: Move it outside

**Review by Slowlife01 — COMMENTED** [review]
~~Are you sure there's no grammar error on those strings~~

File: websites/A/AnimesOnline/presence.ts (hunk 8 above)
Comment (Bas950) [reply websites/A/AnimesOnline/presence.ts:26]: Yea browsing stamp should stay outside.

Comment (Faelayis) [reply websites/A/AnimesOnline/presence.ts:26]: What should I do 😅😅

File: websites/A/AnimesOnline/presence.ts [hunk 9]
```diff
@@ -0,0 +1,124 @@
+const presence = new Presence({
+    clientId: "874161536074145833"
+  }),
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused",
+    browsing: "presence.activity.browsing"
+  });
```
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:8]:
```suggestion
  }),
  browsingStamp = Math.floor(Date.now() / 1000);
```

File: websites/A/AnimesOnline/presence.ts (hunk 8 above)
Comment (Bas950) [inline websites/A/AnimesOnline/presence.ts:26]:
```suggestion
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
Afterwards upload new proof

**Comment by Faelayis** [comment]
> Afterwards upload new proof

image
image
image
image
image
image

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**